test/framework/src/main/java/org/opensearch/test/ExternalTestCluster.java

```diff
@@ -98,6 +102,7 @@ public ExternalTestCluster(
         Function<Client, Client> clientWrapper,
         String clusterName,
         Collection<Class<? extends Plugin>> pluginClasses,
+        Map<Class<? extends Plugin>, Class<? extends Plugin>> extendedPlugins,
```

Comment:
```suggestion
        Collection<? extends PluginInfo> classpathPlugins,
```

One of the reasons we've introduced `PluginInfo` for classpath plugins is to express the dependencies between plugins naturally, using `PluginInfo::extendedPlugins`.

Reply:
@reta this is already supported in [NodeConfigurationSource](https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/NodeConfigurationSource.java#L62-L65), but what is lacking is the ability to define extending relationships between plugins. I think it makes sense to add another method similar to `nodePlugins()` to define those relationships for the classpathPlugins for testing.

Reply:
@cwperks This is my point: the relationships between plugins are defined by using `PluginInfo` (not complimentary maps) and this is how the `Node` deals with `classpathPlugins`.

Reply:
Modified this to allow the NodeConfigurationSource to also return a collection of `PluginInfo` to support testing with extended plugins. Thanks reta, I think this allows the ultimate flexibility.